### ISSUE TITLE
E11000 duplicate key error index - iOS SDK support needed

### ISSUE THREAD

**Issue by bhushpar82** [strongloop/loopback#2049]
Hi Team,

We have encountered with duplicate key error when I am trying to update loopback model via iOS SDK. Can you please help us to understand what is the reason for same. I remember before I was able to update, however now when I am trying to update its not allowing.  Below is the error and code:
We are using MongoDB as back end and id is not auto-generated. Is there some setting at configuration end we need to enable?

 var userList:LBModelRepository = adapter.repositoryWithModelName(modelName)
        
        userList.findById(modelID, success: { (messageRecord) -> Void in
            
            
            var onlineDict = messageRecord.toDictionary()
            onlineDict["MessageStatus"] = "onServer" /// update dictionary here
            
            var transaction:LBModel = userList.modelWithDictionary(onlineDict) // call model with Dictionary
            
            //save the model 
            transaction.saveWithSuccess({ () -> Void in
                
                }, failure: { (error) -> Void in
                    print(error)
                               })
            
            }) { (error) -> Void in
                
        }

Error Domain=AFNetworkingErrorDomain Code=-1011 "Expected status code in (200-299), got 500" UserInfo={NSLocalizedRecoverySuggestion={"error":{"code":11000,"index":0,"errmsg":"E11000 duplicate key error index: Originale_db.Ox-MessageCore.$_id_ dup key: { : \"2B0B9C9D-AB81-4BDD-9632-A33E15D4965C20162518246\" }","op":{"Field1":"","OxFileID":"","OxMessageBody":"This is new test","LinkerKey":"GROUP40E5E8D4-D3EF-4081-B22B-1CFE6D5E7E4F2016119225747","MessageStatus":"onServer","OxMessageType":"text","OxReceiverID":["1168","1177","1188"],"OxReceiverType":"multiuser","OxSenderID":"1168","OxSentTime":"2016-02-05T18:02:46.460Z","OxUpdatedAt":"2016-02-05T18:02:46.461Z","OxSenderName":"1168","_id":"2B0B9C9D-AB81-4BDD-9632-A33E15D4965C20162518246"}}}, NSErrorFailingURLKey=/api/Ox-MessageCores, AFNetworkingOperationFailingURLRequestErrorKey=<NSMutableURLRequest: 0x7fa45b7204e0> { URL: /api/Ox-MessageCores }, AFNetworkingOperationFailingURLResponseErrorKey=<NSHTTPURLResponse: 0x7fa45b700390> { URL: /api/Ox-MessageCores } { status code: 500, headers {
    "Access-Control-Allow-Credentials" = true;
    Connection = "keep-alive";
    "Content-Length" = 653;
    "Content-Type" = "application/json; charset=utf-8";
    Date = "Sat, 06 Feb 2016 02:03:20 GMT";
    Etag = "W/\"28d-nlz0Jbfj6ERvEEYKvVAyGw\"";
    Vary = "Origin, Accept-Encoding";
    "X-Powered-By" = Express;
} }, NSLocalizedDescription=Expected status code in (200-299), got 500}

please let me know how this can be resolved. Thanks a lot for the help

**Comment by bhushpar82** [comment]
Hi Team,

We tried to do a clean install and update it again
But still there is an issue, can you pls provide resolution on above mentioned issue

Best,
Bhushan

**Comment by bhushpar82** [comment]
Hi Guys,

Finally found the work around!
Models which we created, we explicitly specified ID in composer.
But when we tried to perform update, it was throwing above error.

However, when I recreated above model and did not specify ID field system created a field called "id" for me. Using this "id" field as primary key, I was able to edit the records. Not sure, if this is bug or this is the way its configured. If yes, then pls. specify this in docs so other may benefit.

Cheers!

Best,
Bhushan
